### PR TITLE
Adding slx_command module and supporting module_utils.

### PR DESCRIPTION
**Pull request by bigmstone**
##### SUMMARY
This PR adds the slx_command module, tests, and the required slxos module_utils.

- [x] CLIConf Tests
- [x] Terminal Tests
- [x] `module_util` Tests

##### ISSUE TYPE:
 - New Module Pull Request

##### COMPONENT NAME
`slxos_command`

##### ANSIBLE VERSION
```
ansible 2.6.0 (slxos-command 44d69dabf2) last updated 2018/04/03 13:48:17 (GMT -500)
  config file = None
  configured module search path = [u'/Users/mstone/dev/stackstorm/ansible/library']
  ansible python module location = /Users/mstone/dev/stackstorm/ansible/lib/ansible
  executable location = /Users/mstone/dev/stackstorm/ansible/bin/ansible
  python version = 2.7.14 (default, Mar 29 2018, 15:58:25) [GCC 4.2.1 Compatible Apple LLVM 9.0.0 (clang-900.0.39.2)]
```

##### ADDITIONAL INFORMATION

```
⟩ ansible-playbook --ask-pass -i hosts.yml command.yml
SSH password:
ERROR! no action detected in task. This often indicates a misspelled module name, or incorrect module path.

The error appears to have been in '/Users/mstone/dev/stackstorm/ansible-slxos/command.yml': line 8, column 7, but may
be elsewhere in the file depending on the exact syntax problem.

The offending line appears to be:

  tasks:
    - name: run multiple commands on remote nodes
      ^ here

⟩ ansible-playbook --ask-pass -i hosts.yml command.yml
SSH password:

PLAY [dc2leaf2] *********************************************************************************************************

TASK [run multiple commands on remote nodes] *********************************************************************************************************
ok: [dc2leaf2]

PLAY RECAP *********************************************************************************************************
dc2leaf2                   : ok=1    changed=0    unreachable=0    failed=0
```

### CODE REVIEW

File: lib/ansible/modules/network/slxos/slxos_command.py [hunk 1]
```diff
@@ -0,0 +1,245 @@
...
+  wait_for:
+    description:
+      - List of conditions to evaluate against the output of the
+        command. The task will wait for each condition to be true
+        before moving forward. If the conditional is not true
+        within the configured number of retries, the task fails.
+        See examples.
+    required: false
+    default: null
+    aliases: ['waitfor']
```
Comment (gundalow) [inline lib/ansible/modules/network/slxos/slxos_command.py]: This line can be removed, no need to use the old name

File: lib/ansible/modules/network/slxos/slxos_command.py [hunk 2]
```diff
@@ -0,0 +1,245 @@
...
+        the number of retries has expired.
+    required: true
+  wait_for:
+    description:
+      - List of conditions to evaluate against the output of the
+        command. The task will wait for each condition to be true
+        before moving forward. If the conditional is not true
+        within the configured number of retries, the task fails.
+        See examples.
+    required: false
```
Comment (gundalow) [inline lib/ansible/modules/network/slxos/slxos_command.py]: `required: false` lines can be removed, as that's the default

File: lib/ansible/modules/network/slxos/slxos_command.py [hunk 3]
```diff
@@ -0,0 +1,245 @@
...
+    return commands
+
+
+def main():
+    """main entry point for module execution
+    """
+    argument_spec = dict(
+        commands=dict(type='list', required=True),
+
+        wait_for=dict(type='list', aliases=['waitfor']),
```
Comment (gundalow) [inline lib/ansible/modules/network/slxos/slxos_command.py]: Please remove the aliases

File: lib/ansible/modules/network/slxos/slxos_command.py [hunk 4]
```diff
@@ -0,0 +1,245 @@
...
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+
+DOCUMENTATION = """
+---
+module: slxos_command
+version_added: "2.6"
+author: "Lindsay Hill (@LindsayHill)"
+short_description: Run commands on remote devices running SLX-OS
```
Comment (gundalow) [inline lib/ansible/modules/network/slxos/slxos_command.py]: We generally include the company name in the short_description to make it easier for people to find the modules
See http://docs.ansible.com/ansible/latest/modules/list_of_network_modules.html

`Run commands on remote devices running Extreme Networks SLX-OS`

**Review by gundalow — COMMENTED** [review]
Just checking.
Does this *only* support `connection: network_cli`?

We no longer need to support `connection: local` + `provider`

**Comment by bigmstone** [comment]
@gundalow Correct, this is only intended to be used w/ `network_cli`.

**Comment by bigmstone** [comment]
@gundalow I've added tests for `cliconf` file. There wasn't a current test precedent for testing `cliconf` methods so I took some liberties there. If it's going to be a blocker for getting it merged then I'll remove it. If you're 👍 then I'd also like to add `terminal` and `module_util` tests.

**Review by ganeshrn — APPROVED** [review]

**Comment by ganeshrn** [comment]
@bigmstone Please update your Github handle https://github.com/ansible/ansible/blob/devel/.github/BOTMETA.yml for the new files added with this PR

**Comment by bigmstone** [comment]
@ganeshrn Everything should be in place now. From my end I'm 👍for this to be merged.

**Review by gundalow — APPROVED** [review]

**Comment by bigmstone** [comment]
@gundalow @ganeshrn Anything else needed from our end for this to get merged? I have a few other modules to add after this gets merged, and want to make sure I understand Ansible's process in this round to make merging the others smooth.

**Comment by gundalow** [comment]
Merged into `devel` for release in Ansible 2.6.0

You can now raise one PR per module for the remaining modules you wish to add.

**Pull request merged**